File: v19_facade/python/movie1.py
class Projector:
    def on(self):
        print("Turning on the projector...")

    def off(self):
        print("Turning off the projector...")

class DVDPlayer:
    def play(self):
        print("Playing the movie...")

    def stop(self):
        print("Stopping the movie...")

class SoundSystem:
    def on(self):
        print("Turning on the sound system...")

    def off(self):
        print("Turning off the sound system...")

# Client code
projector = Projector()
dvd_player = DVDPlayer()
sound_system = SoundSystem()

projector.on()
dvd_player.play()
sound_system.on()

File: v19_facade/java/Movie1.java
package v19_facade.java;

class Projector {
    public void on() {
        System.out.println("Turning on the projector...");
    }

    public void off() {
        System.out.println("Turning off the projector...");
    }
}

class DVDPlayer {
    public void play() {
        System.out.println("Playing the movie...");
    }

    public void stop() {
        System.out.println("Stopping the movie...");
    }
}

class SoundSystem {
    public void on() {
        System.out.println("Turning on the sound system...");
    }

    public void off() {
        System.out.println("Turning off the sound system...");
    }
}

public class Movie1 {
    public static void main(String[] args) {
        Projector projector = new Projector();
        DVDPlayer dvdPlayer = new DVDPlayer();
        SoundSystem soundSystem = new SoundSystem();

        projector.on();
        dvdPlayer.play();
        soundSystem.on();
    }
}
